src/app/teamSheets/page.tsx

```diff
@@ -0,0 +1,55 @@
+import { redirect } from 'next/navigation'
+
+import { faUsersRectangle } from '@fortawesome/free-solid-svg-icons'
+
+import { getKindeServerSession } from '@kinde-oss/kinde-auth-nextjs/server'
+
+import prisma from '@db/prismaSingleton'
+
+import { findOrCreateUser } from '@functions/user'
+
+import PageHeader from '@components/PageHeader'
+
+import type { TeamSheetWithTeam } from '@types'
+
+import TeamSheetListItem from './_TeamSheetListItem'
+
+const Page = async () => {
+  const { getUser } = getKindeServerSession()
+  const kindeCurrentUser = await getUser()
+
+  const currentUser = await findOrCreateUser(kindeCurrentUser)
+  const { id: currentUserId } = currentUser
+
+  if (!currentUserId) redirect('/#')
+
+  const userTeamSheets = await prisma.teamSheet.findMany({
+    where: {
+      userId: currentUserId,
+    },
+    include: {
```

Comment:
regarding your comment in the other component, i don't see how to do this other than including all those nested entities in the teamSheet request. the only optimisation i can think of it that all the competitions the team is in will be for the same sport, so is it possible to only request the first competitionTeam record, instead of all?

```
include: {
  team: {
    include: {
      competitionTeams: {
        include: {
          competition: {
            include: {
              sport: true
            },
          },
        },
      },
    },
  },
},
```